flake.nix

```diff
@@ -2,36 +2,82 @@
   description = "templ";
 
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs?ref=nixos-unstable";
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    devshell = {
+      url = "github:numtide/devshell";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-parts = {
+      url = "github:hercules-ci/flake-parts";
+    };
     gitignore = {
       url = "github:hercules-ci/gitignore.nix";
       inputs.nixpkgs.follows = "nixpkgs";
     };
   };
 
-  outputs = { self, nixpkgs, gitignore }:
-    let
+  outputs = inputs@{ self, nixpkgs, gitignore, flake-parts, devshell }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        devshell.flakeModule
+        flake-parts.flakeModules.easyOverlay
+      ];
+
       # Systems supported
-      allSystems = [
+      systems = [
         "x86_64-linux" # 64-bit Intel/AMD Linux
         "aarch64-linux" # 64-bit ARM Linux
         "x86_64-darwin" # 64-bit Intel macOS
         "aarch64-darwin" # 64-bit ARM macOS
       ];
 
-      # Helper to provide system-specific attributes
-      forAllSystems = f: nixpkgs.lib.genAttrs allSystems (system: f {
-        pkgs = import nixpkgs { inherit system; };
-      });
-    in
-    {
-      packages = forAllSystems ({ pkgs }: {
-        default = pkgs.buildGoModule {
-          name = "templ";
-          src = gitignore.lib.gitignoreSource ./.;
-          subPackages = [ "./cmd/templ" ];
-          vendorSha256 = "sha256-7QYF8BvLpTcDstkLWxR0BgBP0NUlJ20IqW/nNqMSBn4=";
+      perSystem = { config, pkgs, final, system, ... }: {
+        overlayAttrs = {
+          inherit (config.packages)
+            templ
+            templ-docs;
+        };
+
+        packages = rec {
+          default = templ;
+          templ = pkgs.buildGo121Module rec {
+            name = "templ";
+
+            src = gitignore.lib.gitignoreSource ./.;
+
+            subPackages = [ "cmd/templ" ];
+
+            vendorSha256 = "sha256-7QYF8BvLpTcDstkLWxR0BgBP0NUlJ20IqW/nNqMSBn4=";
+
+            CGO_ENABLED = 0;
+
+            ldflags = [
+              "-s"
+              "-w"
+              "-extldflags -static"
+            ];
+          };
+          templ-docs = pkgs.buildNpmPackage rec {
+            pname = "frontend";
+            version = "0.0.0";
+
+            src = gitignore.lib.gitignoreSource ./docs;
+
+            npmDepsHash = "sha256-/9FzD38s0bER9zvR4AfU/KBrzRsWDmuFeuMgSL45/1s=";
+
+            installPhase = ''
+              mkdir -p $out/share
+              cp -r build/ $out/share/docs
+            '';
+          };
+        };
+
+        devshells.default = {
```

Comment:
I'm not familiar with devshell and why I'd want to use it. Not sure why I'd take a library for this instead of using the built-in stuff?

As per above...

```nix
      devShell = forAllSystems
        ({ system, pkgs }:
          pkgs.mkShell {
            buildInputs = [
              pkgs.gotestsum # Alternative test runner.
              xc.packages.${system}.xc
              templ.packages.${system}.default
            ];
          });
```

Reply:
I like devshell because it gives you some extra features like the ability to create commands, basically I use them as aliases to run other scripts.

It also loads much faster than the standard mkShell. I believe it has some extra caching. This is pretty important to me because I work on a lot of stuff and move around a lot, creating many terminals in the same project, so the dev shell loading quickly is big. But it is not needed, so we can revert to vanilla stuff.